packages/@aws-cdk/aws-lambda/test/function-hash.test.ts

```diff
@@ -36,103 +37,130 @@ describe('function hash', () => {
         handler: 'index.handler',
       });
 
-      expect(calculateFunctionHash(fn1)).toEqual(calculateFunctionHash(fn2));
-      expect(calculateFunctionHash(fn1)).toEqual('aea5463dba236007afe91d2832b3c836');
+      expect(calculateFunctionHash('v1', fn1)).toEqual(calculateFunctionHash('v1', fn2));
+      expect(calculateFunctionHash('v1', fn1)).toEqual('aea5463dba236007afe91d2832b3c836');
     });
-  });
 
-  test('code impacts hash', () => {
-    const stack1 = new Stack();
-    const fn1 = new lambda.Function(stack1, 'MyFunction1', {
-      runtime: lambda.Runtime.NODEJS_12_X,
-      code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
-      handler: 'index.handler',
+    test('code impacts hash', () => {
+      const stack1 = new Stack();
+      const fn1 = new lambda.Function(stack1, 'MyFunction1', {
+        runtime: lambda.Runtime.NODEJS_12_X,
+        code: lambda.Code.fromAsset(path.join(__dirname, 'my-lambda-handler')),
+        handler: 'index.handler',
+      });
+
+      expect(calculateFunctionHash('v1', fn1)).not.toEqual('aea5463dba236007afe91d2832b3c836');
```

Comment:
This check seems pretty redundant given the below line. I guess it's a callback to the test above, but it's a little weird to assert both != A and == B back to back.

Reply:
The weirder thing is - none of this was originally working. Because the hash was previously including the full cloudformation output, the `DependsOn` clause would always be different, since they depended on different roles and hence the hash would be different anyway.

I'll fix these up.